commit e02a294c5d7cea0dcf17652417cc7d1ac3169955 (refs/changes/30/4230/1)
Author: Jared Morrow <jm@joyent.com>
Date:   2018-06-15T08:13:57-06:00 (1 year, 4 months ago)
    
    MANTA-3764 Create manta_fastdelete_queue bucket in moray for fast GC

diff --git a/lib/moray.js b/lib/moray.js
index 99d8f4f..ea22b68 100644
--- a/lib/moray.js
+++ b/lib/moray.js
@@ -69,6 +69,15 @@ var DELETE_LOG_SCHEMA = {
 };
 var DELETE_LOG_VERSION = 1;
 
+var FASTDELETE_QUEUE_BUCKET = process.env.MANTA_FASTDELETE_QUEUE_BUCKET ||
+    'manta_fastdelete_queue';
+var FASTDELETE_QUEUE_SCHEMA = {
+    objectId: {
+        type: 'string'
+    }
+};
+var FASTDELETE_QUEUE_VERSION = 1;
+
 var DIR_COUNT_BUCKET = 'manta_directory_counts';
 var DIR_COUNT_SCHEMA = {
     entries: {
@@ -130,6 +139,16 @@ function setupMantaBuckets(log, client, cb) {
                 }
             },
             log: log
+        }, {
+            client: client,
+            bucket: FASTDELETE_QUEUE_BUCKET,
+            opts: {
+                index: FASTDELETE_QUEUE_SCHEMA,
+                options: {
+                    version: FASTDELETE_QUEUE_VERSION
+                }
+            },
+            log: log
         }, {
             client: client,
             bucket: DIR_COUNT_BUCKET,
